core/src/main/java/io/grpc/Attributes.java

```diff
@@ -152,6 +152,21 @@ public int hashCode() {
     return data.hashCode();
   }
 
+  /**
+   * Merges two attributes objects into one.
+   */
+  public static Attributes merge(Attributes a, Attributes b) {
+    if (a == Attributes.EMPTY) {
```

Comment:
just curious, how bad if we just always do `return Attributes.newBuilder(a).setAll(b).build();`? I guess we can also do something like 

```
if (a == Attributes.EMPTY && b == Attributes.EMPTY) {
  return Attributes.EMPTY;
}
return Attributes.newBuilder(a).setAll(b).build();
```